### PR TITLE
Replace Carrierwave with ActiveStorage

### PR DESCRIPTION
**Pull request by sebastiancaraballo**
**Description:**
This project was using `carrierwave` and `fog-aws` gems for uploading user's avatar. As discussed in [#138](https://github.com/rootstrap/rails_api_base/issues/138), ActiveStorage is the recommended way to work with attachments. 

In order to work with base64 attachments, [`active-storage-base64` gem](https://github.com/rootstrap/active-storage-base64) was used and it works correctly.

**Tasks:**
- Add `aws-sdk-s3` gem for Amazon S3 support.
- Add `active_storage_base64` gem (not released yet) for base64 support.
- Add migration deleting `avatar` column (used by carrierwave before).
- Add migration adding ActiveStorage tables (`rails active_storage:install`).
- Since avatar is not a column anymore, views were modified in order to render avatar's url, accessing  `avatar_url` method. 
- Upgrade documentation.

Feel free to recommend better ways to accomplish this.

Thanks [@santib](https://github.com/santib) for your help.

### CODE REVIEW

File: config/environments/development.rb [hunk 1]
```diff
@@ -1,6 +1,8 @@
 Rails.application.configure do
   # Settings specified here will take precedence over those in config/application.rb.
 
+  routes.default_url_options = { host: 'localhost:3000' }
```
Comment (santib) [inline config/environments/development.rb:4]: I think you are missing this config for production

Comment (santib) [inline config/environments/development.rb:4]: I think you are missing this config for production

**Review by mdornel — APPROVED** [review]
Approved after @santib comments 👍

**Review by jpascual1994 — APPROVED** [review]

File: app/models/application_record.rb [hunk 2]
```diff
@@ -1,3 +1,4 @@
 class ApplicationRecord < ActiveRecord::Base
   self.abstract_class = true
+  include ActiveStorageSupport::SupportForBase64
```
Comment (gabrielbursztein2) [inline app/models/application_record.rb]: In how many models are you using this? Do you think that it's necessary for all models to include this support?

File: app/views/api/v1/registrations/show.json.jbuilder [hunk 3]
```diff
@@ -0,0 +1,3 @@
+json.user do
+  json.partial! 'info', user: @resource
```
Comment (gabrielbursztein2) [inline app/views/api/v1/registrations/show.json.jbuilder]: `json.partial! 'api/v1/registrations/info', user: @resource

File: app/views/api/v1/sessions/show.json.jbuilder [hunk 4]
```diff
@@ -0,0 +1,3 @@
+json.user do
+  json.partial! 'info', user: @resource
```
Comment (gabrielbursztein2) [inline app/views/api/v1/sessions/show.json.jbuilder]: Same here

File: spec/models/user_spec.rb [hunk 5]
```diff
@@ -20,14 +20,11 @@
 #  confirmation_sent_at   :datetime
 #  unconfirmed_email      :string
 #  name                   :string
-#  nickname               :string
```
Comment (gabrielbursztein2) [inline spec/models/user_spec.rb:23]: Is this right?

Comment (sebastiancaraballo) [reply spec/models/user_spec.rb:23]: Yes, this is an annotation from a [previous PR](https://github.com/sebastiancaraballo/target/pull/28).

**Review by gabrielbursztein2 — APPROVED** [review]

**Review by santib — APPROVED** [review]

File: app/models/application_record.rb (hunk 2 above)
Comment (sebastiancaraballo) [reply app/models/application_record.rb]: Right. I'll move this to User model.

**Pull request merged**